src/qdlutils/applications/qdlscan/main.py

```diff
@@ -0,0 +1,1289 @@
+        module = importlib.import_module(import_path)
+        logger.debug(f"Importing {import_path}")
+        constructor = getattr(module, class_name)
+        y_axis = constructor()
+        y_axis.configure(config[APPLICATION_NAME][y_axis_name]['configure'])
+        # Get the limits
+        self.min_y_position = config[APPLICATION_NAME][y_axis_name]['configure']['min_position']
+        self.max_y_position = config[APPLICATION_NAME][y_axis_name]['configure']['max_position']
+        self.max_y_range = self.max_y_position - self.min_y_position
+
+        # Get the z axis instance
+        import_path = config[APPLICATION_NAME][z_axis_name]['import_path']
+        class_name = config[APPLICATION_NAME][z_axis_name]['class_name']
+        module = importlib.import_module(import_path)
+        logger.debug(f"Importing {import_path}")
+        constructor = getattr(module, class_name)
+        z_axis = constructor()
+        z_axis.configure(config[APPLICATION_NAME][z_axis_name]['configure'])
+        # Get the limits
+        self.min_z_position = config[APPLICATION_NAME][z_axis_name]['configure']['min_position']
+        self.max_z_position = config[APPLICATION_NAME][z_axis_name]['configure']['max_position']
+        self.max_z_range = self.max_x_position - self.min_x_position
+
+        # Get the application controller constructor 
+        import_path = config[APPLICATION_NAME]['ApplicationController']['import_path']
+        class_name = config[APPLICATION_NAME]['ApplicationController']['class_name']
+        module = importlib.import_module(import_path)
+        logger.debug(f"Importing {import_path}")
+        constructor = getattr(module, class_name)
+        # Get the configure dictionary
+        controller_config_dict = config[APPLICATION_NAME]['ApplicationController']['configure']
+        # Create the application controller passing the hardware and the config dict
+        # as kwargs.
+        self.application_controller = constructor(
+            **{'x_axis_controller': x_axis,
+               'y_axis_controller': y_axis,
+               'z_axis_controller': z_axis,
+               'counter_controller': counter,
+               **controller_config_dict}
+        )
+
+    def load_yaml_from_name(self, yaml_filename: str) -> None:
+        '''
+        Loads the yaml configuration file from name.
+
+        Should be called during instantiation of this class and should be the callback
+        function for loading of other standard yaml files while running.
+
+        Parameters
+        ----------
+        yaml_filename: str
+            Filename of the .yaml file in the qdlscan/config_files path.
+        '''
+        yaml_path = importlib.resources.files(CONFIG_PATH).joinpath(yaml_filename)
+        self.configure_from_yaml(str(yaml_path))
+
+    def enable_buttons(self) -> None:
+        pass
+    def disable_buttons(self) -> None:
+        pass
+
+    def optimize_x_axis(self, tkinter_event=None) -> None:
```

Comment:
opening the counter here doesn't work for each axis

Reply:
add in gaussian optimization to go to optimized position is worth it

Reply:
Fixed the counter call back. Added a basic gaussian optimization algorithm for optimizing the position. Currently the application is set to use this method as the default, but users can write new methods or set it back to the "none" method if desired.